src/dug/core/async_search.py

```diff
@@ -690,3 +418,134 @@ async def search_kg(self, unique_id, query, offset=0, size=None,
+                            "match": {
+                                "search_terms": {
+                                    "query": query,
+                                    "fuzziness": fuzziness,
+                                    "prefix_length": prefix_length,
+                                    "operator": "and",
+                                    "boost": 5
+                                }
+                            }
+                        },
+                        {
+                            "match": {
+                                "element_desc": {
+                                    "query": query,
+                                    "fuzziness": fuzziness,
+                                    "prefix_length": prefix_length,
+                                    "operator": "and",
+                                    "boost": 3
+                                }
+                            }
+                        },
+                        {
+                            "match": {
+                                "element_desc": {
+                                    "query": query,
+                                    "fuzziness": fuzziness,
+                                    "prefix_length": prefix_length,
+                                    "boost": 2
+                                }
+                            }
+                        },
+                        {
+                            "match": {
+                                "element_name": {
+                                    "query": query,
+                                    "fuzziness": fuzziness,
+                                    "prefix_length": prefix_length,
+                                    "boost": 2
+                                }
+                            }
+                        },
+                        {
+                            "match": {
+                                "search_terms": {
+                                    "query": query,
+                                    "fuzziness": fuzziness,
+                                    "prefix_length": prefix_length,
+                                    "boost": 1
+                                }
+                            }
+                        },
+                        {
+                            "match": {
+                                "optional_terms": {
+                                    "query": query,
+                                    "fuzziness": fuzziness,
+                                    "prefix_length": prefix_length
+                                }
+                            }
+                        }
+                    ]
+                }
+            }
+        }
+        if concept:
+            query['bool']['must'] = {
+                "match": {
+                    "identifiers": concept
+                }
+            }
+        return query
+
+    def get_simple_search_query(self, query):
```

Comment:
I think this function should have some documentation with a reference to https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-simple-query-string-query.html.